feat: add HTML source mode editor for admin users

ledjay opened the pull request:
Hey @kaaloo 👋

Since the editorial team sometimes ends up with messy nested tags from the current text editor, they needed a way to peek “under the hood” and fix things directly.

So… I added a code mode!
And I couldn’t resist adding a bit of syntax highlighting while I was at it 😄

~Needs a few tests to make sure it handles the in-out data perfectly but I'm almost there.~
Made some refactor and few tests, seems all good.

ledjay committed:
feat: add HTML source mode editor for admin users

- Implemented syntax-highlighted source code editor using Prism.js with custom dark theme
- Added toggle button for admins to switch between rich text and HTML source views
- Enhanced source editor with proper indentation, tab handling, and live syntax highlighting

ledjay committed:
refactor: extract source editor into dedicated component

- Moved HTML source editing logic from RichTextInput into standalone SourceEditor component
- Improved cursor position preservation during syntax highlighting updates using character offset tracking
- Enhanced code organization by separating visual/source mode concerns into distinct components

kaaloo reviewed (changes requested):
I don't see any HTML sanitation taking place on the HTML source editor content.

Recommended Security Fix
The handleSourceChange function in 
RichTextInput.tsx
 should sanitize the HTML before storing/processing it:

```typescript
// In RichTextInput.tsx
import { sanitizeContent } from "~/lib/sanitizeContent";

const handleSourceChange = useCallback(
  (html: string) => {
    const sanitizedHtml = sanitizeContent(html);
    setSourceText(sanitizedHtml);
    if (props.onChange) {
      props.onChange(sanitizedHtml);
    }
  },
  [props],
);
```

ledjay committed:
feat: add HTML sanitization to rich text input

- Integrated sanitizeContent function to clean user-generated HTML content
- Applied sanitization to both internal state and onChange callback to prevent XSS vulnerabilities

ledjay commented:
> I don't see any HTML sanitation taking place on the HTML source editor content.
> 
> Recommended Security Fix The handleSourceChange function in RichTextInput.tsx should sanitize the HTML before storing/processing it:
> 
> ```ts
> // In RichTextInput.tsx
> import { sanitizeContent } from "~/lib/sanitizeContent";
> 
> const handleSourceChange = useCallback(
>   (html: string) => {
>     const sanitizedHtml = sanitizeContent(html);
>     setSourceText(sanitizedHtml);
>     if (props.onChange) {
>       props.onChange(sanitizedHtml);
>     }
>   },
>   [props],
> );
> ```

@kaaloo merci pour ton retour.
Vu que cette feature n'est dispo que pour les admins je n'avais pas la sanitization à l'esprit.
Mais en effet, on est jamais trop prudents !
Fixed !

kaaloo commented:
Closing this for now as it's quite stale
